src/renderer/screens/lend/modals/ManageLend/index.js

```diff
@@ -229,4 +208,122 @@ const ManageModal = ({ name, account, parentAccount, ...rest }: Props) => {
   );
 };
 
+const Banner = ({
+  capabilities,
+  locale,
+  onClose,
+  t,
+  onSelectAction,
+  unit,
+  totalSupplied,
+  currency,
+  account,
+  parentAccount,
+}: {
+  capabilities: {
+    canSupply: boolean,
+    canSupplyMax: Boolean,
+    enabledAmount: BigNumber,
+    enabledAmountIsUnlimited: boolean,
+    canSupplyMax: boolean,
+    status: CompoundAccountStatus,
+  },
+  locale: string,
+  unit: Unit,
+  currency: TokenCurrency,
+  onSelectAction: (name: string, onClose: () => void, nextStep?: string, cta?: string) => void,
+  onClose: () => void,
+  t: any,
+  totalSupplied: BigNumber,
+  account: TokenAccount,
+  parentAccount: ?Account,
+}) => {
+  const dispatch = useDispatch();
+  const { enabledAmount, enabledAmountIsUnlimited, status, canSupplyMax } = capabilities;
+
+  const label =
+    enabledAmountIsUnlimited && totalSupplied.eq(0) ? (
+      <Trans i18nKey="lend.manage.enable.approveLess" />
+    ) : !canSupplyMax ? (
+      <Trans i18nKey="lend.manage.enable.approveMore" />
+    ) : status === "ENABLING" ? (
+      <Trans i18nKey="lend.manage.enable.viewDetails" />
+    ) : (
+      <Trans i18nKey="lend.manage.enable.approve" />
+    );
+
+  const text =
+    enabledAmount.gt(0) && enabledAmountIsUnlimited ? (
+      <Trans
+        i18nKey="lend.manage.enable.infoNoLimit"
+        values={{
+          supplied: formatCurrencyUnit(unit, totalSupplied, {
+            locale,
+            showAllDigits: false,
+            disableRounding: false,
+            showCode: true,
+          }),
+        }}
+      >
+        <b></b>
+      </Trans>
+    ) : enabledAmount.gt(0) ? (
+      <Trans
+        i18nKey="lend.manage.enable.info"
+        values={{
+          amount:
+            enabledAmount &&
+            formatCurrencyUnit(unit, enabledAmount, {
+              locale,
+              showAllDigits: false,
+              disableRounding: true,
+              showCode: true,
+            }),
+          assetName: currency.ticker,
+        }}
+      >
+        <b></b>
+      </Trans>
+    ) : enabledAmountIsUnlimited && totalSupplied.eq(0) ? (
+      <Trans i18nKey="lend.manage.enable.notSuppliedNoLimit" />
+    ) : enabledAmount.gt(0) && totalSupplied.eq(0) ? (
+      <Trans i18nKey="lend.manage.enable.notEnabled" />
+    ) : !canSupplyMax ? (
+      <Trans i18nKey="lend.manage.enable.notEnoughApproved" />
+    ) : status === "ENABLING" ? (
+      <Trans i18nKey="lend.manage.enable.enabling" />
+    ) : status === null ? (
+      <Trans i18nKey="lend.manage.enable.notEnabled" />
+    ) : (
+      <div />
+    );
+
+  const action = useCallback(() => {
+    if (status === "ENABLING") {
+      const op = account.pendingOperations.find(o => o.extra?.approving);
```

Comment:
I think the logic here is a bit prone to change in future and therefore we really should move this into live-common helper function 🙏

Reply:
done